--- boost_1_58_0/boost/test/impl/plain_report_formatter.ipp	2012-12-13 13:32:58.000000000 -0800
+++ boost_1_58_0.patched/boost/test/impl/plain_report_formatter.ipp	2015-06-11 19:48:28.000000000 -0700
@@ -62,7 +62,7 @@
                   const_string name, const_string res )
 {
     if( v > 0 ) {
-        ostr << std::setw( indent ) << ""
+        ostr << std::setw( (int)indent ) << ""
              << v << ' ' << name << ( v != 1 ? "s" : "" );
         if( total > 0 )
             ostr << " out of " << total;
@@ -112,7 +112,7 @@
     else
         descr = "failed";
 
-    ostr << std::setw( m_indent ) << ""
+    ostr << std::setw( (int)m_indent ) << ""
          << "Test " << (tu.p_type == tut_case ? "case " : "suite " ) << quote() << tu.p_name << ' ' << descr;
 
     if( tr.p_skipped ) {
