--- boost_1_70_0/boost/thread/pthread/mutex.hpp	2020-08-03 16:01:18.000000000 -0700
+++ boost_1_70_0.patch/boost/thread/pthread/mutex.hpp	2020-08-03 16:10:38.000000000 -0700
@@ -59,7 +59,7 @@
         {
           int const res = posix::pthread_mutex_destroy(&m);
           boost::ignore_unused(res);
-          BOOST_ASSERT(!res);
+          //BOOST_ASSERT(!res);
         }
 
         void lock() BOOST_THREAD_ACQUIRE()
