--- boost_1_55_0/boost/date_time/int_adapter.hpp	2008-11-12 11:37:53.000000000 -0800
+++ boost_1_55_0.patched/boost/date_time/int_adapter.hpp	2014-03-12 14:06:42.000000000 -0700
@@ -230,7 +230,14 @@
         return int_adapter::neg_infinity();
       }
     }
+#ifdef __clang__
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wconversion"
+#endif
     return int_adapter<int_type>(value_ + rhs.as_number());
+#ifdef __clang__
+#pragma clang diagnostic pop
+#endif
   }
 
   int_adapter operator+(const int_type rhs) const
