--- boost_1_81_0/boost/array.hpp	2023-01-14 14:48:10
+++ boost_1_81_0.patched/boost/array.hpp	2023-01-14 14:53:52
@@ -182,7 +182,11 @@
 
         // check range (may be private because it is static)
         static BOOST_CONSTEXPR bool rangecheck (size_type i) {
-            return i >= size() ? boost::throw_exception(std::out_of_range ("array<>: index out of range")), true : true;
+            if(i >= size())
+            {
+                boost::throw_exception(std::out_of_range ("array<>: index out of range"));
+            }
+            return true;
         }
 
     };
