--- boost_1_53_0/boost/atomic/detail/cas64strong.hpp	2012-12-15 10:28:27.000000000 -0800
+++ boost_1_53_0.patched/boost/atomic/detail/cas64strong.hpp	2014-03-11 23:39:41.000000000 -0700
@@ -220,7 +220,7 @@
     {
         value_type original = load(memory_order_relaxed);
         do {
-        } while (!compare_exchange_weak(original, original + v, order, memory_order_relaxed));
+        } while (!compare_exchange_weak(original, (char*)original + v, order, memory_order_relaxed));
         return original;
     }
 
@@ -229,7 +229,7 @@
     {
         value_type original = load(memory_order_relaxed);
         do {
-        } while (!compare_exchange_weak(original, original - v, order, memory_order_relaxed));
+        } while (!compare_exchange_weak(original, (char*)original - v, order, memory_order_relaxed));
         return original;
     }
 
