--- boost_1_53_0/boost/test/unit_test_suite_impl.hpp	2009-07-03 23:36:59.000000000 -0700
+++ boost_1_53_0.patched/boost/test/unit_test_suite_impl.hpp	2014-03-12 14:26:32.000000000 -0700
@@ -74,7 +74,7 @@
     readwrite_property<counter_t>       p_expected_failures;    // number of expected failures in this test unit
     mutable readwrite_property<bool>    p_enabled;              // enabled status for this unit
 
-    void                                increase_exp_fail( unsigned num );
+    void                                increase_exp_fail( counter_t num );
 
 protected:
     ~test_unit();
