--- boost_1_63_0//tools/build/src/tools/clang-darwin.jam	2017-01-12 15:18:28.000000000 -0800
+++ boost_1_63_0.patched//tools/build/src/tools/clang-darwin.jam	2017-01-12 14:54:35.000000000 -0800
@@ -92,7 +92,7 @@
 toolset.flags clang-darwin.compile OPTIONS <inlining>on   : -Wno-inline ;
 toolset.flags clang-darwin.compile OPTIONS <inlining>full : -Wno-inline ;
 
-toolset.flags clang-darwin.compile OPTIONS <warnings>off : -w ;
+toolset.flags clang-darwin.compile OPTIONS <warnings>off : ;
 toolset.flags clang-darwin.compile OPTIONS <warnings>on  : -Wall ;
 toolset.flags clang-darwin.compile OPTIONS <warnings>all : -Wall -pedantic ;
 toolset.flags clang-darwin.compile OPTIONS <warnings-as-errors>on : -Werror ;
